### PR TITLE
Refactor node creation from entities

### PR DESCRIPTION
**Pull request by jeevb**
Related to: https://github.com/flyteorg/flyte/issues/1179

### CODE REVIEW

**Review by kumare3 — COMMENTED** [review]
lgtm

**Comment by jeevb** [comment]
Smoke tested with:
1. No values set for `interruptible` at the task or workflow level - task is not interruptible
2. Task has `interruptible=True`, but workflow has no value set - task is interruptible
3. Workflow has `interruptible=True`, but task has no value set - task is interruptible
4. Workflow has `interruptible=False`, but task has `interruptible=True` - task is interruptible
5. Workflow has `interruptible=True`, but task has `interruptible=False` - task is not interruptible

File: flytekit/core/task.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def task(
     cache: bool = False,
     cache_version: str = "",
     retries: int = 0,
-    interruptable: bool = False,
+    interruptible: Optional[bool] = None,
```
Comment (tvetterli-fn) [inline flytekit/core/task.py:67]: how come we don't keep this as `default=False` (same question for the `TaskMetadata` above?

**Review by tvetterli-fn — APPROVED** [review]

**Review by jganbat-fn — APPROVED** [review]

File: flytekit/core/task.py (hunk 1 above)
Comment (jeevb) [reply flytekit/core/task.py:67]: For this test:
Workflow has `interruptible=True`, but task has no value set - task is interruptible

If the default is `False`, the task does not inherit the value from the workflow. Using `interruptible=False` as a default explicitly tells the system NOT to schedule on interruptible nodes. So in reality there are three states: interruptible, not interruptible, and not set. Does that make sense?

**Pull request merged**